src/Microsoft.Extensions.SecretManager.Tools/Program.cs

```diff
@@ -134,59 +136,37 @@ internal int RunInternal(params string[] args)
                 CommandOutputProvider.LogLevel = LogLevel.Debug;
             }
 
-            var userSecretsId = ResolveUserSecretsId(options);
+            var userSecretsId = !string.IsNullOrEmpty(options.Id)
+                    ? options.Id
+                    : ResolveIdFromProject(options.Project);
+
             var store = new SecretsStore(userSecretsId, Logger);
             options.Command.Execute(store, Logger);
             return 0;
         }
 
-        private string ResolveUserSecretsId(CommandLineOptions options)
+        internal string ResolveIdFromProject(string projectPath)
         {
-            var projectPath = options.Project ?? _workingDirectory;
-
-            if (!Path.IsPathRooted(projectPath))
-            {
-                projectPath = Path.Combine(_workingDirectory, projectPath);
-            }
-
-            if (!projectPath.EndsWith("project.json", StringComparison.OrdinalIgnoreCase))
-            {
-                projectPath = Path.Combine(projectPath, "project.json");
-            }
-
-            var fileInfo = new PhysicalFileInfo(new FileInfo(projectPath));
+            var finder = new GracefulProjectFinder(_workingDirectory);
+            var projectFile = finder.FindMsBuildProject(projectPath);
 
-            if (!fileInfo.Exists)
-            {
-                throw new GracefulException(Resources.FormatError_ProjectPath_NotFound(projectPath));
-            }
-
-            Logger.LogDebug(Resources.Message_Project_File_Path, fileInfo.PhysicalPath);
-            return ReadUserSecretsId(fileInfo);
-        }
+            Logger.LogDebug(Resources.Message_Project_File_Path, projectFile);
 
-        // TODO can use runtime API when upgrading to 1.1
-        private string ReadUserSecretsId(IFileInfo fileInfo)
-        {
-            if (fileInfo == null || !fileInfo.Exists)
+            try
             {
-                throw new GracefulException($"Could not find file '{fileInfo.PhysicalPath}'");
+                var project = new MsBuildProjectContextBuilder()
+                    .UseMsBuild(_msBuildContext)
+                    .AsDesignTimeBuild()
+                    .WithBuildTargets(Array.Empty<string>())
+                    .WithProjectFile(projectFile)
+                    .WithTargetFramework("") // TFM doesn't matter
```

Comment:
This tool doesn't need to worry about TFMs. We just want to read a global property. The alternative would be `.BuildAllTargetFrameworks().First()`